code/modules/ruins/lavaland_ruin_code.dm

```diff
@@ -0,0 +1,213 @@
+	LAVA.start()
+	ongoing_weather = FALSE
+
+/obj/machinery/lavaland_controller/Destroy()
+	return QDEL_HINT_LETMELIVE
+
+
+
+//lavaland_surface_seed_vault.dmm
+//Seed Vault
+
+/obj/effect/spawner/lootdrop/seed_vault
+	name = "seed vault seeds"
+	lootcount = 1
+
+	loot = list(/obj/item/seeds/gatfruit = 10,
+				/obj/item/seeds/cherry = 15,
+				/obj/item/seeds/berry/glow = 10,
+				/obj/item/seeds/sunflower/moonflower = 8
+				)
+
+/obj/effect/landmark/corpse/mob_spawn/seed_vault
+	name = "sleeper"
+	mobname = "Vault Creature"
+	icon = 'icons/obj/Cryogenic2.dmi'
+	icon_state = "sleeper"
+	//mob_species = /datum/species/podman//TODO: Delete this vault or use existing species
+	flavour_text = {"You are a strange, artificial creature. Your creators were a highly advanced and benevolent race, and launched many seed vaults into the stars, hoping to aid fledgling civilizations. You are to tend to the vault and await the arrival of sentient species. You've been waiting quite a while though..."}
+
+/obj/effect/landmark/corpse/mob_spawn/seed_vault/special(mob/living/new_spawn)
+	var/plant_name = pick("Tomato", "Potato", "Brocolli", "Carrot", "Deathcap", "Ambrosia", "Pumpkin", "Ivy", "Kudzu", "Bannana", "Moss", "Flower", "Bloom", "Spore", "Root", "Bark", "Glowshroom", "Petal", "Leaf", "Venus", "Sprout","Cocao", "Strawberry", "Citrus", "Oak", "Cactus", "Pepper")
+	new_spawn.real_name = plant_name
+
+//Greed
+
+/obj/structure/cursed_slot_machine
+	name = "greed's slot machine"
+	desc = "High stakes, high rewards."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "slots-on"
+	anchored = 1
+	density = 1
+	var/win_prob = 5
+/*
+/obj/structure/cursed_slot_machine/attack_hand(mob/living/carbon/human/user)
+	if(!istype(user))
+		return
+	if(in_use)
+		return
+	in_use = TRUE
+	user << "<span class='danger'><B>You feel your very life draining away as you pull the lever...it'll be worth it though, right?</B></span>"
+	user.adjustCloneLoss(20)
+	if(user.stat)
+		user.gib()
+	icon_state = "slots2"
+	sleep(50)
+	icon_state = "slots1"
+	in_use = FALSE
+	if(prob(win_prob))
+		new /obj/item/weapon/dice/d20/fate/one_use(get_turf(src))
+		if(user)
+			user << "You hear laughter echoing around you as the machine fades away. In it's place...more gambling."
+			qdel(src)
+	else
+		if(user)
+			user << "<span class='danger'>Looks like you didn't win anything this time...next time though, right?</span>"
+*/
+//Gluttony
+
+/obj/effect/gluttony
+	name = "gluttony's wall"
+	desc = "Only those who truly indulge may pass."
+	anchored = 1
+	density = 1
+	icon_state = "blob"
+	icon = 'icons/mob/blob.dmi'
+
+/obj/effect/gluttony/CanPass(atom/movable/mover, turf/target, height=0)//So bullets will fly over and stuff.
+	if(height==0)
+		return 1
+	if(istype(mover, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = mover
+		if(H.nutrition >= NUTRITION_LEVEL_FAT)
+			return 1
+		else
+			H << "<span class='danger'><B>You're not gluttonous enough to pass this barrier!</B></span>"
```

Comment:
also you should be using the `to_chat` proc instead of the `<<` operator for displaying messages to a target (in this case H)